src/server/stream_family_test.cc

```diff
@@ -159,6 +159,72 @@ TEST_F(StreamFamilyTest, XRead) {
   EXPECT_THAT(resp, ArgType(RespExpr::NIL_ARRAY));
 }
 
+TEST_F(StreamFamilyTest, XReadGroup) {
```

Comment:
I would to see some more test cases, specifically the following queries fail (I pulled your branch locally and played around with redis-ci):

Test dataset:
```
Create a new stream:
xadd mystream * id 1
>> "1688466177147-0"
xadd mystream * id 2
>> "1688466178855-0"
xadd mystream * id 3
>> "1688466180281-0"

bgsave 
>> OK //Store the snapshot so you can reload the streams on crashes and reproduce the bugs below easier

And now let's test print with xread:

xread STREAMS mystream 1688466177146-0

1) 1) "mystream"
2) 1) 1) "1688466177147-0"
        2) 1) "id"
            2) "1"
    2) 1) "1688466178855-0"
        2) 1) "id"
            2) "2"
    3) 1) "1688466180281-0"
        2) 1) "id"
            2) "3"
```

checks without creating a consumer group via `xgroup create`    
``` 
1. xreadgroup GROUP group_a client_1 NOACK STREAMS mystream > 
    result -> dragonfly segfaults
2. xreadgroup GROUP group_a client_1 COUNT 10 BLOCK 10000 NOACK STREAMS mystream >
    result -> dragonfly segfaults
3. xreadgroup GROUP group_a client_1 STREAMS mystream > 
    result -> dragonfly sefaults
4. xreadgroup GROUP group_a client_1 STREAMS mystream 1688467144385-0 
    result -> NIL (prints an error on redis that consumer group is not created)
```

Reply:
added test cases to address the "group doesn't exist" situations. The error is now fixed